### PR TITLE
Remove correspondences from translation averaging outliers downstream

### PR DESCRIPTION
**Pull request by akshay-krishnan**

### CODE REVIEW

**Comment by akshay-krishnan** [comment]
Based on evaluation on larger datasets, this change produces small improvements (1% Pose AUC post BA) for some datasets (Barn, Kings college, Skansen, Gerrard hall) and small regressions of similar magnitude for others (skydio 501). this also somewhat depends on the frontend being used. 

Some example numbers: 


| dataset |  BA pose AUC with change | BA pose AUC without change |
|---|---|---|
Barn - LOFTR | 22.71 | 22.69 |
Barn - SIFT | 40.94 | 39.39 |
Gerrard Hall - SIFT | 6.36 | 5.90 |
Gerrard Hall - LOFTR | 4.71 | 7.52 |
Kings college  - SIFT | 42.80 | 40.84 |
Kings college - LOFTR | 66.12 | 59.55 | 
Palace - SIFT | 82.45 | 82.88 |
Skansen - SIFT | 87.86 | 87.19 |
Skansen - superglue | 74.92 | 74.73 |
skydio501 - LOFTR | 5.16 | 5.70 |
skydio501 - SIFT | 1.93 | 10.1 |
south building - LOFTR | 44.45 | 44.08 |
south building - SIFT | 49.66 | 48.77 |

more numbers are in the [spreadsheet](https://docs.google.com/spreadsheets/d/1WBME42-CtZH0Vcr1YC2QsSv3cmJpFXBdklAxEc1eFfY/edit#gid=1993353546) (ta-outliers-x)

**Comment by akshay-krishnan** [comment]
requesting high-level thoughts on whether we should proceed with this change. Shall update the code if thats the case.

**Comment by travisdriver** [comment]
What degree threshold are the previous pose AUC statistics at?

**Comment by akshay-krishnan** [comment]
Most of them are at 1 deg, I used 2.5 degrees for a couple (including skydio501) because their pose AUCs at 1 deg are 0. Ive highlighted the rows in the spreadsheet that I used.

File: gtsfm/multi_view_optimizer.py [hunk 1]
```diff
@@ -196,3 +199,16 @@ def get_2d_tracks(
 ) -> List[SfmTrack2d]:
     tracks_estimator = CppDsfTracksEstimator()
     return tracks_estimator.run(corr_idxs_dict, keypoints_list)
+
+
+def filter_corr_by_idx(correspondences, idxs):
```
Comment (johnwlambert) [inline gtsfm/multi_view_optimizer.py:204]: nit: missing type annotations

**Review by ayushbaid — APPROVED** [review]

**Comment by johnwlambert** [comment]
Btw @akshay-krishnan  I believe CI is failing:
```
=========================== short test summary info ============================
FAILED tests/averaging/translation/test_averaging_1dsfm.py::TestTranslationAveraging1DSFM::test_circle_all_edges - ValueError: too many values to unpack (expected 2)
FAILED tests/averaging/translation/test_averaging_1dsfm.py::TestTranslationAveraging1DSFM::test_computation_graph - ValueError: too many values to unpack (expected 2)
FAILED tests/averaging/translation/test_averaging_1dsfm.py::TestTranslationAveraging1DSFM::test_lund_door - ValueError: too many values to unpack (expected 2)
FAILED tests/averaging/translation/test_averaging_1dsfm.py::TestTranslationAveraging1DSFM::test_panorama - ValueError: too many values to unpack (expected 2)
FAILED tests/averaging/translation/test_averaging_1dsfm.py::Test1dsfmAllOutliers::test_outlier_case_missing_value - ValueError: too many values to unpack (expected 2)
FAILED tests/frontend/detector_descriptor/test_d2net.py::TestD2NetDetDesc::test_coordinates_range - EOFError: Ran out of input
FAILED tests/frontend/detector_descriptor/test_d2net.py::TestD2NetDetDesc::test_detect_and_describe_shape - EOFError: Ran out of input
FAILED tests/frontend/detector_descriptor/test_d2net.py::TestD2NetDetDesc::test_number_of_detections - EOFError: Ran out of input
FAILED tests/frontend/detector_descriptor/test_d2net.py::TestD2NetDetDesc::test_pickleable - EOFError: Ran out of input
FAILED tests/frontend/detector_descriptor/test_d2net.py::TestD2NetDetDesc::test_scale - EOFError: Ran out of input
===== 10 failed, 425 passed, 7 skipped, 24 warnings in 1023.58s (0:17:03) ======
```

**Pull request merged**